components/schemes/State/examples/src/MDCColorResource.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+// Note: Make sure to update [semanticResourceFromName:] when changing or adding values
+typedef NS_ENUM(NSInteger, MDCColorResourceSemantic) {
+  MDCColorResourceSemanticNone,
+  MDCColorResourceSemanticPrimary,
+  MDCColorResourceSemanticOnPrimary,
+  MDCColorResourceSemanticSurface,
+  MDCColorResourceSemanticOnSurface,
+  MDCColorResourceSemanticOverlay,
+  MDCColorResourceSemanticBackground,
+  MDCColorResourceSemanticOnBackground,
+  MDCColorResourceSemanticOutline,
+  MDCColorResourceSemanticError,
+};
+
+// Note: Make sure to update [paletteResourceFromName:] when changing or adding values
+typedef NS_ENUM(NSInteger, MDCColorResourcePalette) {
+  MDCColorResourcePaletteNone,
+  MDCColorResourcePaletteGrey,
+  MDCColorResourcePaletteBlue,
+  MDCColorResourcePaletteBlueGrey,
+  MDCColorResourcePaletteIndigo,
+  MDCColorResourcePalettePurple,
+  MDCColorResourcePaletteRed,
+  MDCColorResourcePaletteOrange,
+  MDCColorResourcePaletteYellow,
+};
+
+// Note: Make sure to update [paletteTintResourceFromName:] when changing or adding values
+typedef NS_ENUM(NSInteger, MDCColorResourcePaletteTint) {
+  MDCColorResourcePaletteTintNone,
+  MDCColorResourcePaletteTint50,
+  MDCColorResourcePaletteTint100,
+  MDCColorResourcePaletteTint200,
+  MDCColorResourcePaletteTint300,
+  MDCColorResourcePaletteTint400,
+  MDCColorResourcePaletteTint500,
+  MDCColorResourcePaletteTint600,
+  MDCColorResourcePaletteTint700,
+  MDCColorResourcePaletteTint800,
+  MDCColorResourcePaletteTint900,
+};
+
+static const CGFloat MDCColorResourceDefaultOpacity = 1.0;
```

Comment:
Nit: Constants should start with `k`

http://google.github.io/styleguide/objcguide.html#constants